fix(docs): improve router docs 2 (#23176)
expo/expo: docs/pages/router/reference/static-rendering.mdx

<FileTree files={['dist/blog/alpha.html', 'dist/blog/beta.html']} />

<APIBox header="generateStaticParams">

> A server-only function which is evaluated at build-time in a Node.js environment by Expo CLI. This means it has access to `__dirname`, `process.cwd()`, `process.env`, etc. and every environment variable that's available in the process, not just the values prefixed with **EXPO*PUBLIC***. **generateStaticParams** is not run in a browser environment, so it cannot access browser APIs like **localStorage** or **document**, nor can it access native Expo APIs like **expo-camera** or **expo-location**.

Reviewer: ```suggestion
A server-only function evaluated at build-time in a Node.js environment by Expo CLI. This means it has access to `__dirname`, `process.cwd()`, `process.env`, and more. It also has access to every environment variable that's available in the process, not just the values prefixed with **EXPO*PUBLIC***. **generateStaticParams** is not run in a browser environment, so it cannot access browser APIs like **localStorage** or **document**, nor can it access native Expo APIs such as **expo-camera** or **expo-location**.
```
